### PR TITLE
JBPM-7206: New custom query mapper for processes

### PR DESCRIPTION
**Pull request by mcivantos-tribalyte**
I've added some modifications in the mapper ProcessInstancesCustom and a new test in order to search processes without vars.

### CODE REVIEW

**Comment by tsurdilo** [comment]
ok to test

**Comment by tsurdilo** [comment]
@mcivantos-tribalyte there is not test added with this change. Prob needs to be please. Thanks.

**Comment by mcivantos-tribalyte** [comment]
@tsurdilo  Here, you have the test :)

**Comment by tsurdilo** [comment]
@mcivantos-tribalyte thanks. looks good to me. @mswiderski ?

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/mapper/ProcessInstanceCustomQueryMapper.java [hunk 1]
```diff
@@ -59,7 +59,9 @@ public static ProcessInstanceCustomQueryMapper get() {
                     String varName = getColumnStringValue(dataSetResult, COLUMN_VAR_NAME, i);
                     String varValue = getColumnStringValue(dataSetResult, COLUMN_VAR_VALUE, i);
                     
-                    ((org.jbpm.kie.services.impl.model.ProcessInstanceCustomDesc)pi).addVariable(varName, varValue);
+                    if (varName != null) {
+                        ((org.jbpm.kie.services.impl.model.ProcessInstanceCustomDesc)pi).addVariable(varName, varValue);
```
Comment (jomarko) [inline jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/mapper/ProcessInstanceCustomQueryMapper.java]: Maybe we could apply kiegroup codestyle for better readibility:
```java
((org.jbpm.kie.services.impl.model.ProcessInstanceCustomDesc) pi).addVariable(varName, varValue);
```

File: jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/QueryServiceImplTest.java [hunk 2]
```diff
@@ -390,6 +393,30 @@ public void testGetProcessInstancesCustomWithVars() {
...
+    
+    @Test
+    public void testGetProcessInstancesCustomWithoutVars() {
+
+        query = new SqlQueryDefinition("jbpmProcessSearchWithoutVars", dataSourceJNDIname);
+        query.setExpression("select p. PROCESSINSTANCEID, p.PROCESSID,  p.PROCESSNAME, p.PROCESSVERSION, " +
+                            "p.STATUS,  p.EXTERNALID, pr.STARTDATE,   p.USER_IDENTITY, p.PROCESSINSTANCEDESCRIPTION, " +
+                            "p.CORRELATIONKEY, p.PARENTPROCESSINSTANCEID, pr.LASTMODIFICATIONDATE " +
+                            "from PROCESSINSTANCELOG p " +
+                            "inner join PROCESSINSTANCEINFO pr on p.PROCESSINSTANCEID = pr.INSTANCEID");
```
Comment (jomarko) [inline jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/QueryServiceImplTest.java]: I think there are some not necessary white spaces, could we update as:
```java
query.setExpression("select p.PROCESSINSTANCEID, p.PROCESSID,  p.PROCESSNAME, p.PROCESSVERSION, " +
                    "p.STATUS, p.EXTERNALID, pr.STARTDATE, p.USER_IDENTITY, p.PROCESSINSTANCEDESCRIPTION, " +
                    "p.CORRELATIONKEY, p.PARENTPROCESSINSTANCEID, pr.LASTMODIFICATIONDATE " +
                    "from PROCESSINSTANCELOG p " +
                    "inner join PROCESSINSTANCEINFO pr on p.PROCESSINSTANCEID = pr.INSTANCEID");
```

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/mapper/ProcessInstanceCustomQueryMapper.java (hunk 1 above)
Comment (mcivantos-tribalyte) [reply jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/mapper/ProcessInstanceCustomQueryMapper.java]: Done

File: jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/QueryServiceImplTest.java (hunk 2 above)
Comment (mcivantos-tribalyte) [reply jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/QueryServiceImplTest.java]: Done

**Pull request merged**